src/main/java/com/google/neighborgood/servlets/EditTaskServlet.java

```diff
@@ -20,27 +20,94 @@
 import com.google.appengine.api.datastore.EntityNotFoundException;
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Transaction;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.neighborgood.helper.RewardingPoints;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that handles the request for editing the details and rewards for a certain task. */
+/** Servlet that handles the request for editing tasks. */
 @WebServlet("/tasks/edit")
 public class EditTaskServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
     String keyString = request.getParameter("task-id");
-    int rewardPts = getRewardingPoints(request, "reward-input");
-    if (rewardPts == -1) {
+    Key taskKey = KeyFactory.stringToKey(keyString);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Entity task;
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (!userService.isUserLoggedIn()) {
+      System.err.println("User must be logged in to edit a task");
+      response.sendError(
+          HttpServletResponse.SC_UNAUTHORIZED,
+          "You must be logged in to perform this action on a task");
+    }
+
+    // Edits tasks that have been claimed by setting the "helper" property to the userId
+    // of the helper and changing the task's status to "IN PROGRESS"
+    if (request.getParameterMap().containsKey("action")) {
+      if (request.getParameter("action").equals("helpout")) {
+
+        // Makes use of Transactions to prevent race condition
+        Transaction transaction = datastore.beginTransaction();
+
+        try {
+          task = datastore.get(taskKey);
```

Comment:
Should this use the pattern shown here?  https://cloud.google.com/datastore/docs/concepts/transactions#datastore-datastore-transactional-update-java   So here, this would be transaction.get(taskKey).  

Here's their sample code I'm looking at:

```
Entity task;
Transaction txn = datastore.newTransaction();
try {
  task = txn.get(taskKey);
  if (task == null) {
    task = Entity.newBuilder(taskKey).build();
    txn.put(task);
    txn.commit();
  }
} finally {
  if (txn.isActive()) {
    txn.rollback();
  }
}
```

Thoughts?

Reply:
I implemented it that way first as well, but then I realized that we're using the AppEngine Datastore API, which differs slightly. This is the documentation I followed: https://cloud.google.com/appengine/docs/standard/java/datastore/transactions
I think I followed the pattern in those examples, but let me know if I missed anything :)

Reply:
Ok, I don't really understand the part you mentioned about the AppEngine Datastore API, but I see that this code matches the pattern in the link you provided, so LGTM.  :-)

Reply:
Ah my bad. We’ve been using AppEngine to run our web apps during our internship and there’s a specific API for that that includes the use of Datastore within AppEngine.
Sounds good!